java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -0,0 +1,131 @@
+package co.worklytics.psoxy;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import com.google.cloud.secretmanager.v1.*;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.google.protobuf.ByteString;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+
+@Log
+@RequiredArgsConstructor
+public class SecretManagerConfigService implements ConfigService {
+
+    final String namespace;
+
+    final String projectId;
+
+    final Duration defaultTtl;
```

Comment:
`@NonNull`s on these? (maybe doesn't matter for `namespace`, but presumably others can't be `null`